docs/cli-reference/dfx-extension.md

```diff
@@ -0,0 +1,83 @@
+# dfx extension
+
+Use the `dfx extension` command to manage the extensions available in the `dfx` tool. Extensions can provide additional functionality and commands to the `dfx` tool, enhancing its capabilities.
+
+## Basic Usage
+
+```bash
+dfx extension [command] [option] [flag]
+```
+
+## Commands
+
+| Command                                  | Description                     |
+|------------------------------------------|---------------------------------|
+| [`install`](#dfx-extension-install)      | Installs an extension.          |
+| [`uninstall`](#dfx-extension-uninstall)  | Uninstalls an extension.        |
+| [`run`](#dfx-extension-run)              | Executes an extension.          |
+| [`list`](#dfx-extension-list)            | Lists all installed extensions. |
+
+
+## dfx extension install
+
+**Installing an Extension**:
+
+```bash
+dfx extension install sns
+```
+
+This command will install the `sns` extension.
+
+
+## dfx extension list
+
+**Listing Installed Extensions**:
+
+```bash
+dfx extension list
+```
+
+This command will display all the installed extensions.
+
+## dfx extension uninstall
+
+**Uninstalling an Extension**:
+
+```bash
+dfx extension uninstall sns
+```
+
+This command will uninstall the `sns` extension.
+
+## dfx extension run
+
+**Running an Extension**:
```

Comment:
Should we call out the fact that `dfx extension run` is not the recommended way to call an extension?